rest_framework/routers.py

```diff
@@ -52,12 +52,27 @@ def __init__(self):
     def register(self, prefix, viewset, basename=None):
         if basename is None:
             basename = self.get_default_basename(viewset)
-        self.registry.append((prefix, viewset, basename))
+
+        if not self.basename_already_registered(basename):
+            self.registry.append((prefix, viewset, basename))
 
         # invalidate the urls cache
         if hasattr(self, '_urls'):
             del self._urls
 
+    def basename_already_registered(self, new_basename):
+        """
+        If `basename` is already registered, raise an exception
+        """
+        for route in self.registry:
+            prefix, viewset, basename = route
+            if new_basename == basename:
+                msg = (f'Route with basename "{new_basename}" is already registered. '
+                       f'Please provide a unique basename for viewset "{viewset}"')
+                raise ImproperlyConfigured(msg)
+
+        return False
```

Comment:
It would be  better if this method just returns a boolean and the exception is raised by the `register` method.

This will make the method easier to test and also could allow users to use this method for they needs.

The expansion of route could be done directly in the `for` statement.

PS: for better returning a boolean it could be used the [`any`](https://docs.python.org/3/library/functions.html#any) built-in like:

```python
any(basename == new_basename for _prefix, _vieset, basename in self.registry)
```

Reply:
Done